### DISCUSSION TITLE
Getting Ouput only after external CLI is finished

### DISCUSSION THREAD

**Discussion by lamah85** [Tyrrrz/CliWrap#217] [Q&A]
Hello,

currently i try to implement an Console Application which uses CliWrap to call an cli to execute a flashing process for a fpga.

Evertything works great except that i am getting the whole output from the external cli only after the process is fnished. When i use the cli directly i get an information about the current state every couple seconds.

Using CliWrap i captured the output to a stringbuilder. Every line of the output is terminated with \r\n.

I call CliWrap like this:

```
var cmd = await Cli.Wrap(@$"{configModel.PgrCmdExecutablePath}")
            .WithArguments(args => args
                .Add("-infile")
                .Add(configModel!.FpgaXcfFilePath))
            .WithStandardOutputPipe(PipeTarget.ToStringBuilder(buffer))
            .ExecuteAsync(cts.Token);
```

Afterwards the buffer contains the following:
"Lattice Programmer Diamond (64-bit) 3.12.1.454.2 Command Line
Argument check - OK
Loading Programmer Device Database... Done.
Opening XCF file... Done.
Programming XCF Contents... Connected to Lattice Cable Server.
Done.
Lattice Diamond Programmer has exited successfully."

I get the same result using ListenAsync().

What am i doing wrong?

Greetz,
lamah

**Comment by Tyrrrz** [comment]
Does `ListenAsync()` trigger all events only after the process has finished?

You can try wrapping a terminal (e.g. `cmd`) and run the command inside that, to see if that helps.

**Reply by lamah85** [reply]
Thank you for your reply.

Yes, all events get triggered when the process has finished.

Could you be so kind and provide a minimal example for your suggestion?

**Reply by Tyrrrz** [reply]
Something like this:

```csharp
Cli.Wrap("cmd").WithArguments(new[]
{
	"/c",
	new ArgumentsBuilder()
		.Add(configModel.PgrCmdExecutablePath)
		.Add("infile")
		.Add(configModel!.FpgaXcfFilePath)
		.Build()
});
```
